libs/extended-verification/src/Verification.service.ts

```diff
@@ -0,0 +1,192 @@
+  isNotYetValid: boolean;
+  isRevoked: boolean;
+  ebsiTrustedIssuer: boolean;
+}
+
+export interface PresentationVerificationResult {
+  signature: {
+    isValid: boolean;
+    errors: string[];
+  };
+  isExpired: boolean;
+  isNotYetValid: boolean;
+}
+
+export interface VerificationResult {
+  verified: boolean;
+  details: {
+    credentials: CredentialVerificationResult[];
+    presentation: PresentationVerificationResult | null;
+  };
+}
+
+export interface VerificationServiceInitOptions {
+  providers?: Record<string, Provider>;
+}
+
+export class VerificationService {
+  private static veramoAgent: Agent;
+
+  static async init(options?: VerificationServiceInitOptions): Promise<void> {
+    this.veramoAgent = await createVeramoAgent();
+  }
+
+  static async verify(
+    data: W3CVerifiablePresentation | W3CVerifiableCredential,
+    options?: { ebsiChecks?: boolean }
+  ): Promise<Result<VerificationResult>> {
+    const { ebsiChecks } = options || { ebsiChecks: false };
+
+    const verificationResult: VerificationResult = {
+      verified: true,
+      details: {
+        credentials: [],
+        presentation: null,
+      },
+    };
+
+    let presentation: VerifiablePresentation | null = null;
+    const credentialsToVerify: VerifiableCredential[] = [];
+    try {
+      // Try to decode data as Veramo supported presentation format
+      try {
+        if (typeof data === 'string') {
+          presentation = normalizePresentation(data);
+        } else {
+          presentation = data as VerifiablePresentation;
+        }
+      } catch (error) {
+        // We ignore this error as we now know its not a Veramo supported presentation format
+      }
+
+      // If data is a presentation we verify it
+      if (
+        presentation &&
+        presentation.type &&
+        presentation.type.includes('VerifiablePresentation')
+      ) {
+        // Presentation must contain at least one credential
+        if (!presentation.verifiableCredential) {
+          return ResultObject.error(
+            'Presentation does not contain any credentials'
+          );
+        }
+
+        // Verify the presentation
+        const result = await this.veramoAgent.verifyPresentation({
+          presentation,
+        });
+
+        const errorMessage = result.error?.message || '';
+
+        if (!result.verified) {
+          verificationResult.verified = false;
+        }
+
+        // TODO: Check dates
+        verificationResult.details.presentation = {
+          signature: {
+            isValid: result.verified,
+            errors: [
+              // errorMessage,
+            ],
+          },
+          isExpired: errorMessage.includes('expired'),
+          isNotYetValid: errorMessage.includes('nbf'),
+        };
+
+        // Extract credentials from the presentation
+        for (const credential of presentation.verifiableCredential) {
+          credentialsToVerify.push(decodeCredentialToObject(credential));
```

Comment:
I would rather use corresponding methods from the same library.
Either:
`decodeCredentialToObject` and `decodePresentationToObject` from `@veramo/utils`
or
`normalizeCredential` and `normalizePresentation` from `did-jwt-vc`

Related to lines 77 and 132 in this file.